src/reanimated2/js-reanimated/index.ts

```diff
@@ -46,6 +46,12 @@ global._scheduleOnJS = () => {
   );
 };
 
+global._scheduleOnBackgroundQueue = () => {
+  throw new Error(
+    '[Reanimated] _scheduleOnBackgroundQueue should never be called in JSReanimated.'
```

Comment:
1. Three times is not that much, but yeah, we could do it
2. This particular part of code needs a small refactor since there's way more functions that shouldn't be called in JSReanimated mode. We could define all such functions in an array and loop through it.